client.go

```diff
@@ -0,0 +1,287 @@
+package rendezvous
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	pb "github.com/libp2p/go-libp2p-rendezvous/pb"
+
+	ggio "github.com/gogo/protobuf/io"
+	host "github.com/libp2p/go-libp2p-host"
+	inet "github.com/libp2p/go-libp2p-net"
+	peer "github.com/libp2p/go-libp2p-peer"
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+)
+
+var (
+	DiscoverAsyncInterval = 2 * time.Minute
+)
+
+type Rendezvous interface {
```

Comment:
comments on the interface would be helpful

Reply:
yes indeed! Part of the "docstrings" checkbox.